src/app/dynamic-forms/models/cinchy-form.model.ts

```diff
@@ -70,302 +82,402 @@ export class Form implements IForm {
   private _rowId: number;
 
 
+  /**
+   * Returns a shallow copy of the sections contained within this form. Because it is not referential, this set should be considered read-only
+   * unless the whole set is being replaced. Any mutation done directly on the array will not persist.
+   *
+   * To modify values on or underneath the set, use the Form model's mutator functions.
+   */
+  get sections(): Array<FormSection> {
+
+    return this._sections?.slice();
+  }
+  protected set sections(value: Array<FormSection>) {
+
+    this._sections = value.slice();
+  }
+  private _sections = new Array<FormSection>();
+
+
   constructor(
-    public id: string,
-    public name: string,
-    public targetTableId: number,
-    public targetTableDomain: string,
-    public targetTableName: string,
-    public isAccordion: boolean,
-    public hasAccess: boolean,
-    public isChild: boolean = false,
-    public flatten: boolean = false,
-    public childFormParentId?: string,
-    public childFormLinkId?: string,
-    public childFormFilter?: string,
-    public childFormSort?: string
+    public readonly id: string,
+    public readonly name: string,
+    public readonly targetTableId: number,
+    public readonly targetTableDomain: string,
+    public readonly targetTableName: string,
+    public readonly isAccordion: boolean,
+    public readonly hasAccess: boolean,
+    public readonly isChild: boolean = false,
+    public readonly flatten: boolean = false,
+    public readonly childFormParentId?: string,
+    public readonly childFormLinkId?: string,
+    public readonly childFormFilter?: string,
+    public readonly childFormSort?: string,
+    public readonly isClone?: boolean
   ) {}
 
 
   /**
-   * Adds an item to a link array, which takes the form of the value followed by a 0 element. Will not add the value
-   * if it is falsey or if it is already in the set.
+   * Modifies the flattened child form records by either updating an existing entry that matches the given rowId or adding a new entry if no match is present
    */
-  addLinkArrayItem(set: Array<any>, value: string): Array<any> {
+  addOrModifyFlattenedChildRecord(sectionIndex: number, recordData: { [columnName: string]: any }): void {
```

Comment:
Adding comment here to track pending discussion on what flattening child forms means